qiskit/quantum_info/operators/symplectic/base_pauli.py

```diff
@@ -385,6 +385,9 @@ def _from_array(z, x, phase=0):
         if base_z.shape != base_x.shape:
             raise QiskitError("z and x vectors are different size.")
 
+        if num_qubits is not None and base_z.shape[1] != num_qubits:
+            raise ValueError("Provided num_qubits does not match input data.")
```

Comment:
Since this is checking aspects of how this private method is called I would more say this likes like an internal error - I wonder whether QiskitError is more applicable. I would hope any values an end user plugged in did not get this far!